controllers/chat/initChatFromPost.js

```diff
@@ -0,0 +1,76 @@
+const {User} = require('../../databases/models');
+const {StreamChat} = require('stream-chat');
+const {v4: uuid} = require('uuid');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const {responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+const UsersFunction = require('../../databases/functions/users');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const initChatFromPost = async (req, res) => {
+  const {targetUserId} = req.body;
+
+  let members = [targetUserId];
+  if (!members.includes(req.userId)) members.push(req.userId);
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    const [userModel, targetUserModel] = await Promise.all([
+      UsersFunction.findUserById(User, req?.userId),
+      UsersFunction.findUserById(User, targetUserId)
+    ]);
+    /**
+     * @type {import('stream-chat').OwnUserResponse}
+     */
+    const user = {
+      name: userModel?.username,
+      id: req.userId,
+      image: userModel?.profile_pic_path,
+      username: userModel?.username
+    };
+
+    await client.connectUser(user, req.token);
+
+    const channel = client.channel('messaging', uuid(), {
+      members
+    });
+    const createdChannel = await channel.create();
+
+    try {
+      if (!channel?.data?.name) {
+        let channelType = CHANNEL_TYPE.CHAT;
+        if (userModel.is_anonymous || targetUserModel.is_anonymous) {
+          channelType = CHANNEL_TYPE.ANONYMOUS;
+        }
+
+        await channel.updatePartial({
+          set: {
+            channel_type: channelType,
+            name: [userModel?.username, targetUserModel?.username].join(', ')
+          }
+        });
+      }
+    } catch (e) {
+      console.log(e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code is updating the channel after it has been created. If possible, consider setting these properties when creating the channel to reduce the number of operations and potential points of failure.

58:
Avoid using `console.log` for error logging in production code. Use a proper logging library that can provide more context about the error and handle log rotation.



<!-- This is an auto-generated comment by OSS CodeRabbit -->